hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -283,7 +257,7 @@ public void checkpointCoordinator(long checkpointId, CompletableFuture<byte[]> r
     executor.execute(
         () -> {
           try {
-            result.complete(new byte[0]);
+            result.complete(SerializationUtils.serialize(this.eventBuffers.getAllEvents()));
```

Comment:
Maybe we should filter events by: `ckpId < checkpointId - 1`, since events with ckp-id as `checkpointId - 1` are persisted in write state, and will committed later during `notifyCheckpointComplete`, and also

Reply:
The ckpId should already been `< checkpointId - 1`, we have the state check before recommitting.

Reply:
Yes, there is no correctness issue here, and it's about duplicate checkpointing. The `ckpId` actually satisfies `< checkpointId`, since there may exists eager flushing in writers.